### PR TITLE
cilium: bpf macros are no longer used

### PR DESCRIPTION
**Pull request by jrfastab**
The compiler (LLVM) has become smarter, the verifier more able and
Cilium no longer uses these _amazing_ macros. So lets remove them nearly two years later. Convincing C preproccessor to deal with recursion was fun.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7597)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

File: bpf/bpf_overlay.c [hunk 1]
```diff
@@ -199,7 +199,7 @@ static inline int handle_ipv4(struct __sk_buff *skb)
 #endif
 }
 
-__section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_FROM_LXC) int tail_handle_ipv4(struct __sk_buff *skb)
+__section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_FROM_OVERLAY) int tail_handle_ipv4(struct __sk_buff *skb)
```
Comment (joestringer) [inline bpf/bpf_overlay.c]: Where is this defined?

**Review by joestringer — COMMENTED** [review]
RIP

File: bpf/bpf_overlay.c (hunk 1 above)
Comment (jrfastab) [reply bpf/bpf_overlay.c]: yeah popped this off for now will think a bit about it.

**Comment by jrfastab** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

**Comment by borkmann** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Pull request closed without merging**